### ISSUE TITLE
Improve local-exec provisioner documentation

### ISSUE THREAD

**Issue by mikalai-t** [hashicorp/terraform#23502]
### Current Terraform Version
```
Terraform v0.12.15
```

### Use-cases
```hcl
Error: Unsupported block type

  on main.tf line 57, in resource "null_resource" "build":
  57:     environment {

Blocks of type "environment" are not expected here. Did you mean to define
argument "environment"? If so, use the equals sign to assign it a value.
```

### Proposal
Remove confusing word "block" from the `local-exec` provisioner docs:
```markdown
environment - (Optional) **block** of key value pairs ...
```
Fix `inherits` case as it's the first word in the sentence or may be there a first word is missing:
```markdown
...environment of the executed command. **inherits** the current process environment.
```

**Comment by danieldreier** [comment]
@mikalai-t Thank you for suggesting this! That's a great idea. Are you comfortable making a pull request to make this improvement?

**Comment by mikalai-t** [comment]
Yes, but this is the first time I will participate in such kind of projects. )) Could you explain in two words how to do it: clone the repo, checkout a new branch, make changes, push back to this repo and create a new pull request!? How to link pull request with this issue?
Thank you.

**Comment by apparentlymart** [comment]
Hi @mikalai-t! It looks like we lost track of this issue after you responded... sorry about that.

If you're still interested in opening a pull request, you might find it helpful to refer to [our contribution guide](https://github.com/hashicorp/terraform/blob/master/.github/CONTRIBUTING.md), which describes both some technical steps to getting set up and some information on the process we try to follow. You can link to this issue from your pull request just by mentioning its issue number like this: `#23502`. GitHub automatically creates links for mentions like that, for example: #23502.
